index.md

```diff
@@ -38,122 +70,646 @@ Below are the __four scrum ceremonies__ that occur in each sprint.
+
+### Autoboxing
+
+Autoboxing is the automatic conversion that the Java compiler makes between the primitive types and their corresponding object wrapper classes. For example, converting an int to an Integer, a double to a Double, and so on. If the conversion goes the other way, this is called unboxing.
+
+### Autowiring
+
+The Spring framework enables automatic dependency injection. In other words, by declaring all the bean dependencies in a Spring configuration file, Spring container can autowire relationships between collaborating beans. This is called Spring bean autowiring.
+
+### Availability Zone
+
+In the context of cloud computing, it is a public cloud provider’s data center that contains its own power and network connectivity. There are typically multiple Availability Zones in a region.  Each region is a separate geographic area, and each region generally has multiple, isolated locations known as Availability Zones.  A common misconception is that a single zone equals a single data center. In fact, each zone is usually backed by one or more physical data centers, with the largest backed by as many as five.  While a single availability zone can span multiple data centers, no two zones can share a data center. Abstracting things further, to distribute resources evenly across the zones in a given region, some providers independently map zones to identifiers for each account. This means an east coast availability zone for one account may not be backed by the same data centers as an east coast zone for another account.
 
 ## B
 
-__Bill Of Materials (BOM)__
+### Bean
+
+Beans are the objects of classes that are managed by Spring. Traditionally, objects used to create their own dependencies, but Spring manages all the dependencies of an object and instantiates the object after injecting the required dependencies. The @Component annotation is the most common method of defining beans.
+
+### BeanFactory
+
+The BeanFactory is the actual container which instantiates, configures, and manages a number of beans. These beans typically collaborate with one another, and thus have dependencies between themselves. These dependencies are reflected in the configuration data used by the BeanFactory (although some dependencies may not be visible as configuration data, but rather be a function of programmatic interactions between beans at runtime).
+
+### Bill Of Materials (BOM)
+
+A bill of materials (BOM) is a comprehensive inventory of the raw materials, assemblies, subassemblies, parts and components, as well as the quantities of each, needed to manufacture a product. In a nutshell, it is the complete list of all the items that are required to build a product.
+
+### Builder Pattern
```

Comment:
Provide some code example or add link to someother page which provides examples.

<ExternalLink href="https://howtodoinjava.com/design-patterns/creational/builder-pattern-in-java/">Builder Pattern example</ExternalLink>